services/helper/mentees.js

```diff
@@ -1,34 +1,251 @@
+                statusCode: httpStatusCode.ok,
+                message: apiResponses.SESSION_FETCHED_SUCCESSFULLY,
+                result: result,
+                meta: {
+                    type: "feedback",
+                    data: feedbackData.result
+                }
+            });
+        } catch (error) {
+            throw error;
+        }
+    }
+
+    static joinSession(sessionId, token) {
+        return new Promise(async (resolve, reject) => {
             try {
+                const mentee = await userProfile.details(token);
+
+                if (mentee.data.responseCode !== "OK") {
+                    return resolve(common.failureResponse({
+                        message: apiResponses.USER_NOT_FOUND,
+                        statusCode: httpStatusCode.bad_request,
+                        responseCode: 'CLIENT_ERROR'
+                    }));
+                }
+
+                const session = await sessionData.findSessionById(sessionId);
+
+                if (!session) {
+                    return resolve(common.failureResponse({
+                        message: apiResponses.SESSION_NOT_FOUND,
+                        statusCode: httpStatusCode.bad_request,
+                        responseCode: 'CLIENT_ERROR'
+                    }));
+                }
+
+                if (session.status !== "live") {
+                    return resolve(common.failureResponse({
+                        message: apiResponses.JOIN_ONLY_LIVE_SESSION,
+                        statusCode: httpStatusCode.bad_request,
+                        responseCode: 'CLIENT_ERROR'
+                    }));
+                }
 
-                /**
-                 * Your business logic here
-                 */
+                let menteeDetails = mentee.data.result;
+
+                const sessionAttendee =
+                    await sessionAttendees.findLinkBySessionAndUserId(
+                        menteeDetails._id,
+                        sessionId
+                    );
+
+                if (!sessionAttendee) {
+                    return resolve(common.failureResponse({
+                        message: apiResponses.USER_NOT_ENROLLED,
+                        statusCode: httpStatusCode.bad_request,
+                        responseCode: 'CLIENT_ERROR'
+                    }));
+                }
+
+                let link = "";
+                if (sessionAttendee.link) {
+                    link = sessionAttendee.link;
+                } else {
+                    const attendeeLink = await bigBlueButton.joinMeetingAsAttendee(
+                        sessionId,
+                        menteeDetails.name,
+                        session.menteePassword
+                    );
+
+                    await sessionAttendees.updateOne({
+                        _id: sessionAttendee
+                    }, {
+                        link: attendeeLink,
+                        joinedAt: new Date(),
```

Comment:
@aman-tunerlabs lets save joinedAt in UTC format

Reply:
@aman-tunerlabs How we are going to track leftAt when mentee leaves the session ?

Reply:
@aman-g202  Not required of leftAt as of now